### DISCUSSION TITLE
Solarman avec ME3000 SP

### DISCUSSION THREAD

**Discussion by thedarkside33** [StephanJoubert/home_assistant_solarman#683] [Q&A]
Bonjour à tous.

Je débute avec home assistant.

j'ai réussi à installer et configurer Solarman.

J'ai bien les remontés d'informations de mon onduleur et de ma batterie.

La communication se fait via le dongle wifi de l'onduleur 

Mais je souhaiterai pouvoir par exemple mettre l'onduleur en mode stand by ou bien modifier les règles de charge de la batterie sur le réseau etc.

Mais la je sèche. Est ce possible?

Merci d'avance à tous de vos réponses.

**Comment by davidrapan** [comment]
Bonjour @thedarkside33, grâce à mon intégration [davidrapan/ha-solarman](https://github.com/davidrapan/ha-solarman), il est possible de définir des entités qui contrôleront les registres mentionnés. Contactez-moi dans les discussions si vous décidez de l'essayer et si c'est le cas, je peux vous aider à étendre le profil.

**Reply by thedarkside33** [reply]
oui je veux bien de votre aide. Merci

**Reply by thedarkside33** [reply]
Vous parlez Français?

j'ai installé et j'arrive a changer le statut de l'onduleur. Super

Peut on également, changer les règles de charge de la batterie ?

**Reply by davidrapan** [reply]
Quel profil avez-vous utilisé?

**Reply by thedarkside33** [reply]
sofar_wifikit.yaml

**Reply by davidrapan** [reply]
J'ai donc besoin que vous exécutiez l'ACTION intégrée et que vous lisiez le numéro de registre 4609. Vous allez donc dans `Developer tools >> ACTIONS >> Solarman: Read Holding Registers (Modbus Function Code 3)`, sélectionnez votre appareil, définissez le registre sur 4609 et dites-moi quel est le résultat. Et peut-être pourrai-je ajouter un contrôle supplémentaire pour votre modèle d'onduleur.

**Reply by thedarkside33** [reply]
Ok. J’essaye de vous faire ça demain. Merci beaucoup

**Reply by thedarkside33** [reply]
Bonjour il m'affiche: "4609": 30

J'arrive à changer le mode de l'onduleur ( auto, passif ) c'est super merci beaucoup.

Pensez vous que l'on puisse pouvoir choisir les modalités de travail >>modalité de charge puis, pouvoir activer ou désactiver les règles?

Encore merci pour votre travail.

**Reply by davidrapan** [reply]
J'en ai déjà ajouté quelques-uns dans [feat: Time of Use for Sofar profile](https://github.com/davidrapan/ha-solarman/commit/619c300ad9ccf73e3d62c4cdf979f254662da864), alors attendez un peu la nouvelle version ou vous pouvez faire une mise à jour manuelle.

**Reply by thedarkside33** [reply]
Je vais attendre la nouvelle version. La mise à jour se fait toute seule? Je débute avec home assistant. Encore merci pour votre travail.

**Reply by davidrapan** [reply]
Le HA vous avertira lorsqu'une nouvelle version sera disponible. 😉

**Reply by thedarkside33** [reply]
Bonjour,

j'ai mis à jour avec la version 24.12.22

les paramètres de l'onduleur et le changement de mode de travail fonctionne
correctement.( c'est vraiment super)

pour les modes de charge ca s'affiche mais je n'arrive pas à faire ce que
je veux avec et j'ai un message d'erreur ( copie en pièce jointe)

Vous trouverez également une copie d'écran pour vous montrez comment ca
s'affiche.

je vous ai fais un PDF avec ce que je trouve qui serait bien.

Une ligne par programme de charge que l'on active quand on souhaite et
dessous chaque ligne les paramètres que l'on souhaite pour le programme
concerné.

Enfin parlez vous Français ou bien dois je vous écrire dans une autre
langue?

Encore merci pour tout le travail que vous faites car c'est vraiment super.

Bonnes fêtes

Cordialement

Sébastien
![Affichage mode charge](https://github.com/user-attachments/assets/0dfdf98b-7d82-4e9b-93dc-5794eedd7fc6)
![Message d'erreur](https://github.com/user-attachments/assets/6783b14a-3524-4b1b-a120-f4c91e3225c0)
[possibilite d'evolution.pdf](https://github.com/user-attachments/files/18265925/possibilite.d.evolution.pdf)

**Reply by davidrapan** [reply]
Honnêtement, je n'ai aucune idée de ce que sont les codes de fonction Modbus 13 et 14...

Il n'a pas de registres distincts pour des programmes séparés, vous devez d'abord sélectionner le programme que vous souhaitez contrôler à l'aide du numéro, puis vous pouvez utiliser le commutateur et d'autres paramètres, puis vous enregistrez/confirmez les modifications à l'aide de ce bouton de réglage.
